DEVEL.rst

```diff
@@ -8,132 +8,176 @@ If something does not work as expected, please create an issue or
+   $ . devel/bin/activate
+
 
 Continue executing the following commands in the virtualenv you just created.
 
 
-## Install Commissaire
-```
-cd commissaire
-pip install -e .
-popd
-```
+Install Commissaire
+===================
 
+.. code-block::
 
-### Install Commissaire Service
-```
-cd commissaire-service
-pip install -e .
-tools/etcd_init.sh
-```
+   $ cd commissaire
+   $ pip install -e .
+   $ popd
 
-Edit your storage configuration to point to your etcd instance
-```
-cp conf/storage.conf mystorage.conf
-```
-Note: point server_url to http://127.0.0.1:2379 (not https)
+
+Install Commissaire Service
+---------------------------
+
+.. code-block::
+
+   $ cd commissaire-service
+   $ pip install -e .
+   $ tools/etcd_init.sh
+
+
+Edit your storage configuration to point to your etcd instance:
+
+.. code-block::
+
+   $ cp conf/storage.conf mystorage.conf
+
+
+**Note**: Point ``server_url`` to ``http://127.0.0.1:2379`` (not https)
 
 
 Start the service
-```
-commissaire-storage-service -c mystorage.conf &
-popd
-```
 
-### Install Commissaire Server
-```
-cd commissaire-http
-pip install -e .  # Install commissaire-http into the virtualenv
-```
+.. code-block::
+
+   $ commissaire-storage-service -c mystorage.conf &
+   $ popd
+
+
+Install Commissaire Server
+--------------------------
+
+.. code-block::
+
+   $ cd commissaire-http
+   $ pip install -e .  # Install commissaire-http into the virtualenv
+
 
 Edit the configuration to point to your redis instance
-```
-cp conf/commissaire.conf config.conf
-```
-Note: if locally installed you do not need to change anything
 
-Start the service
-```
-commissaire-server -c config.conf &
-popd
-```
+.. code-block::
+
+   $ cp conf/commissaire.conf config.conf
+
+**Note:** If locally installed you do not need to change anything
+
+Start the service:
+
+.. code-block::
+
+   $ commissaire-server -c config.conf &
+   $ popd
+
+
+Run testcases for Commissaire Server
+++++++++++++++++++++++++++++++++++++
+
 
-#### Run testcases for Commissaire Server
-Note that you can use `tox` to run testcases for this project.
+**Note**: that you can use ``tox`` to run testcases for this project.
 
-Install using
+Install using:
 
-```
-pip install tox
-```
+.. code-block::
 
-and then, from the `commissaire-http` folder, run the following
+   $ pip install tox
+
+
+and then, from the ``commissaire-http`` folder, run the following
 command:
 
-```
-tox -v -e py35
-```
+.. code-block:: shell
+
+   $ tox -v -e py35
+
+
+Verification
+------------
+
 
+After this the API will be available at ``http://127.0.0.1:8000/``. To
+verify it works, we will use the initial user ``a`` with pass ``a``.
 
-## Verification
-After this the API will be available at `http://127.0.0.1:8000/`. To
-verify it works, we will use the initial user `a` with pass `a`.
+.. code-block::
 
-```
-curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/
-```
+   $ curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/
```

Comment:
This does indeed fail as well. 

```
(devel) [~/rhat/commissaire/commissaire-http] [1] 10:44:23  (master)
$ curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/                                     
Exception raised in handler <commissaire_http.handlers.JSONRPC_Handler object at 0x7f8b5bacc128>:
Traceback (most recent call last):
  File "/home/tbielawa/rhat/commissaire/commissaire-http/src/commissaire_http/dispatcher/__init__.py", line 180, in dispatch
    return handler(environ, start_response)
...
  File "/home/tbielawa/rhat/commissaire/devel/lib/python3.5/site-packages/kombu/transport/virtual/base.py", line 714, in _lookup
    self.get_table(exchange),
  File "/home/tbielawa/rhat/commissaire/devel/lib/python3.5/site-packages/kombu/transport/redis.py", line 817, in get_table
    raise InconsistencyError(NO_ROUTE_ERROR.format(exchange, key))
kombu.exceptions.InconsistencyError: 
Cannot route message for exchange 'commissaire': Table empty or key no longer exists.
Probably the key ('_kombu.binding.commissaire') has been removed from the Redis database.


127.0.0.1 - - [04/Apr/2017 10:44:24] "GET /api/v0/clusters/ HTTP/1.1" 500 21
```

Cause? Not sure. Maybe because `commissaire-storage-service` isn't running? But those error messages looks like there's an issue with redis. Which is also running

```
(devel) [~/rhat/commissaire/commissaire-http] [1] 10:46:01  (master)
$ sudo lsof -i -P | grep redis                               
redis-ser  8606      redis    4u  IPv4 168935      0t0  TCP deepfryer.redhat.com:6379 (LISTEN)
```

Which is what my `config.conf` is set to

```
(devel) [~/rhat/commissaire/commissaire-http] [1] 10:46:29  (master)
$ grep 6379 ./config.conf                                      
    "bus-uri": "redis://127.0.0.1:6379/",
```